src/sagemaker/cli/compatibility/v2/modifiers/image_uris.py

```diff
@@ -0,0 +1,161 @@
+    "factorization-machine": "factorization-machines",
+    "forecasting-deepar": "forecasting-deepar",
+    "image-classification": "image-classification",
+    "image-classification-neo": "image-classification-neo",
+    "ipinsights": "ipinsights",
+    "kmeans": "kmeans",
+    "knn": "knn",
+    "lda": "lda",
+    "linear-learner": "linear-learner",
+    "ntm": "ntm",
+    "object2vec": "object2vec",
+    "object-detection": "object-detection",
+    "pca": "pca",
+    "randomcutforest": "randomcutforest",
+    "semantic-segmentation": "semantic-segmentation",
+    "seq2seq": "seq2seq",
+    "sagemaker-xgboost": "xgboost",
+    "xgboost-neo": "xgboost-neo",
+}
+
+logger = logging.getLogger("sagemaker")
+
+
+class ImageURIRetrieveRefactor(Modifier):
+    """A class to refactor *get_image_uri() method."""
+
+    def node_should_be_modified(self, node):
+        """Checks if the ``ast.Call`` node calls a function of interest.
+
+        This looks for the following calls:
+
+        - ``sagemaker.get_image_uri``
+        - ``sagemaker.amazon_estimator.get_image_uri``
+        - ``get_image_uri``
+
+        Args:
+            node (ast.Call): a node that represents a function call. For more,
+                see https://docs.python.org/3/library/ast.html#abstract-grammar.
+
+        Returns:
+            bool: If the ``ast.Call`` instantiates a class of interest.
+        """
+        return matching.matches_name_or_namespaces(
+            node, GET_IMAGE_URI_NAME, GET_IMAGE_URI_NAMESPACES
+        )
+
+    def modify_node(self, node):
+        """Modifies the ``ast.Call`` node to call ``image_uris.retrieve`` instead.
+        And switch the first two parameters from (region, repo) to (framework, region)
+
+        Args:
+            node (ast.Call): a node that represents a *image_uris.retrieve call.
+        """
+        original_args = [None] * 3
+        for kw in node.keywords:
+            if kw.arg == "repo_name":
+                arg = kw.value.s
+                modified_arg = ALGORITHM_NAME_FROM_REPO[arg]
+                original_args[0] = ast.Str(modified_arg)
+            elif kw.arg == "repo_region":
+                original_args[1] = ast.Str(kw.value.s)
+            elif kw.arg == "repo_version":
+                original_args[2] = ast.Str(kw.value.s)
+
+        if len(node.args) > 0:
+            original_args[1] = ast.Str(node.args[0].s)
+        if len(node.args) > 1:
+            arg = node.args[1].s
+            modified_arg = ALGORITHM_NAME_FROM_REPO[arg]
+            original_args[0] = ast.Str(modified_arg)
+        if len(node.args) > 2:
+            original_args[2] = ast.Str(node.args[2].s)
+
+        args = []
+        for arg in original_args:
+            if arg:
+                args.append(arg)
+
+        if matching.matches_name(node, GET_IMAGE_URI_NAME) or matching.matches_attr(
+            node, GET_IMAGE_URI_NAME
+        ):
+            node_components = list(pasta.dump(node).split("."))
```

Comment:
using `pasta` to check the attributes strikes me as a bit odd

Reply:
updated using func.value to check for "sagemaker"